package.json

```diff
@@ -32,41 +32,38 @@
   },
   "homepage": "https://github.com/axios/axios",
   "devDependencies": {
-    "bundlesize": "^0.17.0",
     "coveralls": "^3.0.0",
     "es6-promise": "^4.2.4",
-    "grunt": "^1.0.2",
+    "grunt": "^1.3.0",
     "grunt-banner": "^0.6.0",
     "grunt-cli": "^1.2.0",
     "grunt-contrib-clean": "^1.1.0",
     "grunt-contrib-watch": "^1.0.0",
-    "grunt-eslint": "^20.1.0",
-    "grunt-karma": "^2.0.0",
+    "grunt-eslint": "^23.0.0",
+    "grunt-karma": "^4.0.0",
     "grunt-mocha-test": "^0.13.3",
     "grunt-ts": "^6.0.0-beta.19",
-    "grunt-webpack": "^1.0.18",
+    "grunt-webpack": "^4.0.2",
     "istanbul-instrumenter-loader": "^1.0.0",
     "jasmine-core": "^2.4.1",
-    "karma": "^1.3.0",
-    "karma-chrome-launcher": "^2.2.0",
-    "karma-coverage": "^1.1.1",
-    "karma-firefox-launcher": "^1.1.0",
+    "karma": "^5.2.3",
+    "karma-chrome-launcher": "^3.1.0",
+    "karma-firefox-launcher": "^2.1.0",
     "karma-jasmine": "^1.1.1",
     "karma-jasmine-ajax": "^0.1.13",
-    "karma-opera-launcher": "^1.0.0",
     "karma-safari-launcher": "^1.0.0",
-    "karma-sauce-launcher": "^1.2.0",
     "karma-sinon": "^1.0.5",
-    "karma-sourcemap-loader": "^0.3.7",
-    "karma-webpack": "^1.7.0",
+    "karma-sourcemap-loader": "^0.3.8",
+    "karma-webpack": "^4.0.2",
     "load-grunt-tasks": "^3.5.2",
     "minimist": "^1.2.0",
-    "mocha": "^5.2.0",
+    "mocha": "^8.2.1",
     "sinon": "^4.5.0",
-    "typescript": "^2.8.1",
+    "terser-webpack-plugin": "^4.2.3",
+    "typescript": "^4.0.5",
     "url-search-params": "^0.10.0",
-    "webpack": "^1.13.1",
-    "webpack-dev-server": "^1.14.1"
+    "webpack": "^4.44.2",
```

Comment:
Maybe use Webpack 5 since it's out?

ICYMI: took a stab at this in #2536 (and also included removal of superfluous polyfills).

Reply:
There was a conflict with the peer dependencies of [grunt-webpack](https://github.com/webpack-contrib/grunt-webpack/blob/master/package.json#L24). So I rolled back to ^4.

Reply:
@avindra I did see #2536. I'm trying to compartmentalize some of the updates so they are easier to digest and move forward. This PR is larger than I would like, but the interdependency of grunt-karma-webpack in this project makes breaking it down further impossible.

Reply:
@timemachine3030 thanks for having a look. My PR got too big, I will probably cut the polyfill removal stuff to a new PR since it can be shipped separately.

Karma and the sauce stuff is cool for e2e testing but I think there may be merit in whittling these down to plain unit tests (can remove many deps and simplify CI/CD). I don't have any concrete ideas yet about how such a switch should be done, or if it may be too much of a risk for losing the real E2E browser scenarios.